src/tests/ftest/util/security_test_base.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/python
+"""
+  (C) Copyright 2020 Intel Corporation.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. 8F-30005.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+"""
+
+import os
+import random
+from general_utils import pcmd
+
+class DaosTestError(Exception):
+    """DAOS API exception class."""
+
+def acl_entry(usergroup, name, perm, permissions):
+    """Create a daos acl entry for the specified user or group and permission
+
+    Args:
+        usergroup (str): user or group.
+        name (str): user or group name to be created.
+        permission (str): permission to be created.
+
+    Return:
+        str: daos pool acl entry.
+
+    """
+    if perm == "random":
+        perm = random.choice(permissions)
+    if perm == "nonexist":
+        return ""
+    if "group" in usergroup:
+        entry = "A:G:" + name + "@:" + perm
+    else:
+        entry = "A::" + name + "@:" + perm
+    return entry
+
+
+def acl_principal(usergroup, name):
+    """Create a daos ace principal for the specified user or group.
+
+    Args:
+        usergroup (str): user or group.
+        name (str): user or group name to be created.
+
+    Return:
+        str: daos acl entry.
+
+    """
+    if "group" in usergroup:
+        entry = "g:" + name + "@"
+    else:
+        entry = "u:" + name + "@"
+    return entry
+
+
+def add_del_user(hosts, ba_cmd, user):
+    """Add or delete the daos user and group on host by sudo command.
+
+    Args:
+        hosts (list): list of host.
+        ba_cmd (str): linux bash command to create user or group.
+        user (str): user or group name to be created or cleaned.
+
+    """
+    bash_cmd = os.path.join("/usr/sbin", ba_cmd)
```

Comment:
(style)  'ba' may be misspelled - perhaps 'by'?

Reply:
"ba" was for "bash". I changed it from "ba_cmd" to "bash_cmd".
Done